src/schema/v2/previewSavedSearch/searchCriteriaLabel.ts

```diff
@@ -108,7 +108,8 @@ export const resolveSearchCriteriaLabels = async (
     majorPeriods,
     colors,
     partnerIDs,
-  } = parent
+    // Converting all keys to camel case because they can come from Gravity's REST or GraphQL APIs.
+  } = camelCaseKeys(parent) as any
```

Comment:
Before, we didn't have to convert the keys because Gravity's stitched GraphQL response already returned them like this. Now, we're not stitching and using the criteria from Gravity's new REST endpoint.

Let me know if you have any good ideas to make this code a bit cleaner. 🧹

Reply:
Good catch! I've updated the function to correctly convert IDs and added some test cases:

```
export const camelCaseKeys = (
  obj: Record<string, any>
): Record<string, any> => {
  return mapKeys(obj, (_, key) => {
    // Special case for ID to not be converted to I_D
    if (key.includes("_id")) {
      return camelCase(key.replace(/id/g, "ID"))
    }

    return camelCase(key)
  })
}
```